drivers/ethernet/eth_stm32_hal.c

```diff
@@ -191,40 +309,73 @@ static int eth_tx(const struct device *dev, struct net_pkt *pkt)
 
 	heth = &dev_data->heth;
 
-	k_mutex_lock(&dev_data->tx_mutex, K_FOREVER);
-
 	total_len = net_pkt_get_len(pkt);
-	if (total_len > ETH_TX_BUF_SIZE) {
+	if (total_len > (ETH_TX_BUF_SIZE * ETH_TXBUFNB)) {
 		LOG_ERR("PKT too big");
-		res = -EIO;
-		goto error;
+		return -EIO;
 	}
 
+	k_mutex_lock(&dev_data->tx_mutex, K_FOREVER);
+
+#if defined(CONFIG_ETH_STM32_HAL_API_V2)
+
```

Comment:
Nitpick: looking to existing style inside this file, no spaces are needed around these lines.